File: iclusion-importer/src/main/java/com/hartwig/hmftools/iclusion/api/IclusionResponseAdapter.java
package com.hartwig.hmftools.iclusion.api;

import java.util.List;
import java.util.Map;

import com.google.common.collect.Lists;
import com.squareup.moshi.FromJson;

import org.jetbrains.annotations.NotNull;

public class IclusionResponseAdapter {

    @FromJson
    @NotNull
    List<IclusionIndication> indications(@NotNull Map<String, IclusionIndication> json) {
        System.out.println("converting indications");
        return Lists.newArrayList(json.values());
    }

    @FromJson
    @NotNull
    List<IclusionGene> genes(@NotNull Map<String, IclusionGene> json) {
        return Lists.newArrayList(json.values());
    }

    @FromJson
    @NotNull
    List<IclusionVariant> variants(@NotNull Map<String, IclusionVariant> json) {
        return Lists.newArrayList(json.values());
    }

    @FromJson
    @NotNull
    List<IclusionStudy> studies(@NotNull Map<String, IclusionStudy> json) {
        return Lists.newArrayList(json.values());
    }
}

File: api-clients/src/main/kotlin/com/hartwig/hmftools/apiclients/iclusion/data/IclusionResponseAdapter.kt
package com.hartwig.hmftools.apiclients.iclusion.data

import com.squareup.moshi.FromJson

object IclusionResponseAdapter {
    @FromJson
    fun indications(json: Map<String, IclusionIndication>): List<IclusionIndication> {
        return json.values.toList()
    }

    @FromJson
    fun genes(json: Map<String, IclusionGene>): List<IclusionGene> {
        return json.values.toList()
    }

    @FromJson
    fun variants(json: Map<String, IclusionVariant>): List<IclusionVariant> {
        return json.values.toList()
    }

    @FromJson
    fun studies(json: Map<String, IclusionStudy>): List<IclusionStudy> {
        return json.values.toList()
    }
}
